src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
-    console.log(x);
+import { APP_DATA_TYPES } from '../../../config/appDataTypes';
+import { useAppDataContext } from '../../context/AppDataContext';
+import { TEMPLATES } from '../read/cvForm/constants';
+import FirstTemplate from '../read/cvForm/templates/FirstTemplate';
+import DataFilter from './DataFilter';
+import { CandidateRateObj } from './types';
+
+const BuilderView: FC = () => {
+  const { postAppData, patchAppData, appDataArray } = useAppDataContext();
+  const handlePatch = (
+    dataObj: AppData['id'],
+    newData: CandidateRateObj,
+  ): void => {
+    patchAppData({ id: dataObj, data: newData });
   };
-  // todo: implement the logic
-  const handleView = (): void => {
-    const x = 'Viewed';
-    // eslint-disable-next-line no-console
-    console.log(x);
+  const handlePost = (newdata: CandidateRateObj): void => {
+    postAppData({ data: newdata, type: APP_DATA_TYPES.CANDIDATE_RATE_INFO });
+  };
+  const groupedData = appDataArray.groupBy((data) => data.type).toJS();
+  const [selectedType, setSelectedType] = useState('');
+  const handleTypeChange = (value: string): void => {
+    setSelectedType(value);
+  };
+  const dataArray: any[] = Object.keys(groupedData)
+    .filter((key) => key === selectedType)
+    .map((key) => groupedData[key]);
+  const dimensions = [
+    { value: 'mock_type', label: 'Mock Info' },
+    { value: 'personalInfo', label: 'Personal Info' },
+    { value: 'educationInfo', label: 'Education Info' },
+    { value: 'workInfo', label: 'Work Info' },
+    { value: 'skillsInfo', label: 'Skills Info' },
+    { value: 'projectsInfo', label: 'Projects Info' },
+    { value: 'motivationInfo', label: 'Self-Motivation' },
+    { value: 'referencesInfo', label: 'References Info' },
+    { value: 'cvStatusInfo', label: '' },
+  ];
+
+  // Construct the 'cv' object with filtered and grouped data
+  const candidatesCv: { [key: string]: { [key: string]: any } } = {};
+
+  dimensions.forEach((dimension) => {
+    const type = dimension.value;
+    const dataObj = Object.keys(groupedData)
+      .filter((key) => key === type)
+      .map((key) => groupedData[key]);
+
+    const typeDataByMemberId = DataFilter({
+      dataObject: dataObj,
+      targetType: type,
+    });
+
+    Object.keys(typeDataByMemberId).forEach((memberId) => {
+      if (memberId in candidatesCv) {
+        candidatesCv[memberId][type] = typeDataByMemberId[memberId].map(
+          (item) => item.data,
+        );
+      } else {
+        candidatesCv[memberId] = {
+          [type]: typeDataByMemberId[memberId].map((item) => item.data),
+        };
+      }
+    });
+  });
+
+  const [ratings, setRatings] = useState<{ [key: string]: number | null }>({});
+  const handleRatingChange = (
+    memberId: string,
+    newRating: number | null,
+  ): void => {
+    const rateData = appDataArray.find(
+      (data) =>
+        data.data.memberId === memberId && data.type === 'candidateRateInfo',
+    );
```

Comment:
Careful, admins (reviewers) have access to **all** appData, so also the rating from other reviewers. You should also check `data.creator === currentMemberId`. You can get the current member id from the `useLocalContext` hook.

Reply:
I'm trying to acces useLocalContext hook, but it seems not found out here. Also you are right, I'm trying to calculate the mean of the ratings for the same candidate, since there might be more than one reviewer for the same candidate.

Reply:
We might need to take a look, if think the project needs to be updated with the new dependencies. The hook should be accessible from the `@graasp/query-client` package